### ISSUE TITLE
Deprecate `quarto_publish_*()` in favour of rsconnect functions

### ISSUE THREAD

**Issue by hadley** [quarto-dev/quarto-r#102]
Now that rsconnect provides a friendlier UI.  Please feel free to ping me in you think there's anything missing in rsconnect that I can add.

**Comment by hadley** [comment]
I don't have super strong feelings here, except that we should consider the publishing ecosystem holistically, rather than building a bunch of disparate pieces that don't fit together.

**Comment by cderv** [comment]
Yeah, I understand. I am ok to PR into **rsconnect** to bring the specific Quarto support we have here. 
Would that be ok? This way, we could remove those functions here. 

Regarding `quarto publish`, it has some providers that **rsconnect** does not support, and probably won't. So we can't really merge this piece into another one, except maybe for working on a more generic CLI tool different from R / Python packages.
